app/test_main.py

```diff
@@ -1 +1,51 @@
-# write your code here
+from unittest import mock
+import pytest
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url,expected_result,has_internet,valid_url",
+    [
+        pytest.param(
+            "https://www.google.com",
+            "Not accessible",
+            False,
+            True,
+            id="Not accessible without internet"
+        ),
+        pytest.param(
+            "https://www.google.com",
+            "Not accessible",
+            False,
+            False,
+            id="Not accessible when invalid url and no internet"
+        ),
+        pytest.param(
+            "https://www.google.com",
+            "Not accessible",
+            True,
+            False,
+            id="Not accessible when invalid url"
+        ),
+        pytest.param(
+            "https://www.google.com",
+            "Accessible",
+            True,
+            True,
+            id="Accessible when has internet and valid url"
+        ),
+    ]
+)
```

Comment:
Its all good but you can shorten it using this type of approach:
`@pytest.mark.parametrize("name",[value],ids=[id])`